param/__init__.py

```diff
@@ -1840,14 +1844,28 @@ class Foldernames(Foldername):
       is None).
     """
 
-    def _resolve(self, paths: FlexPath):
-        if isinstance(paths, (str, pathlib.Path)):
-            return super()._resolve(paths)
-        elif isinstance(paths, (list, tuple)):
-            return [self._resolve(path) for path in paths]
+    @staticmethod
+    def _cast_to_list(obj):
+        r"""Cast an iterable onto a list, or insert the non-iterable into a one-item list.
+
+        Validation is deferred to _resolve().
+        """
+        if obj is None:
+            return obj
+        elif hasattr(obj, "__iter__") and not isinstance(obj, str):  # store items in  list
+            return list(obj)
```

Comment:
I believe `list(obj)` will give you a different object than `obj`, which means that people can no longer add items to the underlying list they provided. Generally Param does _not_ copy the provided object in cases like this, so I'd prefer to remove this support for iterables in general and ask the user to cast to list if they are working with a non-list iterable. That way the actual original list will be what gets stored here.

Reply:
Initialization now by passing the reference instead of instantiating a `list`